index.js

```diff
@@ -23,8 +28,7 @@ module.exports = function (input, length) {
 
 	while (i--) {
 		var x = pathParts[i];
-
-		if (remainingLength < x.length) {
+		if (remainingLength < x.length + 1) {
```

Comment:
`fn('http://example.com/a/b/cd', 24)` returns `http://example.com/…/b/cd` of length 25.
We need `x.length +1` because of the below code:
```js
remainingLength -= x.length + 1;
```